src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -206,10 +182,21 @@ public Set<String> runBinaryScanner(Set<String> currentFeatureSet, List<String>
                     unavailableFeatures.addAll(getUnavailableMPFeatures(scannerException));
                     throw new FeatureUnavailableException(conflicts, unavailableFeatures, targetMicroProfile,
                             targetJavaEE);
-                } else {
-                    debug("Exception from binary scanner.", scannerException);
-                    throw new PluginExecutionException("Error scanning the application for Liberty features: " + scannerException.toString());
+                } else if (scannerException.getClass().getName().contains("java.lang.IllegalArgumentException")) {
+                    String msg = scannerException.getMessage();
+                    if (msg.contains("CWMIG12056E")) {
+                        if (msg.contains("targetJavaEE")) {
+                            throw new PluginExecutionException("The Java EE or Jakarta EE version number specified in the build file " +
+                                    "is not supported for feature generation.");
+                        } else if (msg.contains("targetMicroProfile")) {
+                            throw new PluginExecutionException("The MicroProfile version number specified in the build file " +
+                                    "is not supported for feature generation.");
```

Comment:
Suggestion to define these messages as constants so that they can be referenced in the integration test without defining new constants.
See https://github.com/OpenLiberty/ci.maven/blob/8b3da42bd86195596d2d7dddabba840b951d9691/liberty-maven-plugin/src/it/generate-features-it/src/test/java/net/wasdev/wlp/test/dev/it/GenerateFeaturesTest.java#L180-L182

Reply:
Good idea.